src/com/google/enterprise/adaptor/DocIdSender.java

```diff
@@ -471,6 +469,33 @@ private String alternateGroupSource(String source) {
     return last;
   }
 
+  private void cleanupGroups(String feedSourceName,
```

Comment:
This is a fair chunk of replicated code to avoid going through pushSizedBatchOfGroups. What's the win?